katcp/test/test_server.py

```diff
@@ -1014,13 +1014,39 @@ def test_sampling(self):
         self.server.ioloop.add_callback(self.server.clear_strategies, client_conn)
         self.server.sync_with_ioloop()
         self.client.assert_request_succeeds("sensor-sampling", "an.int",
-                                            args_equal=[b"an.int", b"none"])
-
+                                            args_equal=["an.int", "none"])
         # Check that we did not accidentally clobber the strategy datastructure
         # in the proccess
         self.client.assert_request_succeeds(
             "sensor-sampling", "an.int", "period", 0.125)
 
+    def test_assert_request_fails(self):
+        self.client.assert_request_fails("sensor-sampling", "an.nonexistentsensor")
+        self.client.assert_request_fails(
+            "sensor-sampling",
+            "an.nonexistentsensor",
+            error_equals="Unknown sensor name: an.nonexistentsensor.",
+        )
+        self.client.assert_request_fails(
+            "sensor-sampling", "an.nonexistentsensor", status_equals="fail"
+        )
+
```

Comment:
Similar to the `assert_request_suceeds` method, it would be good to test that an exception is raised in the negative case.  I.e., if a request that succeeds is sent via `assert_request_fails `.

Reply:
Ok, I was asking if I should throw exceptions in the `assert_request_suceeds` as there is it throws no exceptions. It raises nothing at the moment other than the exceptions you will encounter  in `assertEqual`